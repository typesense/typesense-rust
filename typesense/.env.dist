URL=http://localhost:8108
API_KEY=VerySecretKey