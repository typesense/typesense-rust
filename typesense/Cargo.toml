[package]
name = "typesense"
version = "0.1.0"
authors = ["Typesense  <contact@typesense.org>"]
edition = "2018"

[features]
default = ["derive"]
tokio-rt = ["hyper/runtime", "hyper/tcp", "hyper-tls"]

# Provide derive(Document) macro.
derive = ["typesense_derive"]

[lib]
crate-type = ["cdylib", "rlib"]

[dependencies]
async-trait = "0.1.50"
base64 = "0.13.0"
hmac = "0.11.0"
http = "0.2.4"
serde = { version = "1", features = ["derive"] }
serde_json = "1"
sha2 = "0.9.5"
thiserror = "1.0.24"
typesense_derive = { path="../typesense_derive", optional = true }

[dev-dependencies]
serde_json = "1.0"
trybuild = "1.0.42"

[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
hyper = { version = "0.14.7", features = ["http1", "http2", "client"] }
hyper-tls = { version = "0.5.0", optional = true }

[target.'cfg(target_arch = "wasm32")'.dependencies]
js-sys = { version = "0.3.50" }
wasm-bindgen = { version = "0.2.73" }
wasm-bindgen-futures = { version = "0.4.23" }
web-sys = { version = "0.3.50", features = ["Headers", "Response", "Request", "RequestInit", "RequestMode", "Window", "WorkerGlobalScope"] }

[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies]
tokio = { version = "1.5.0", features = ["macros", "rt", "rt-multi-thread"] }

[target.'cfg(target_arch = "wasm32")'.dev-dependencies]
console_error_panic_hook = "0.1.6"
wasm-bindgen-test = "0.3.23"

[[test]]
name = "derive_tests"
path = "tests/derive/lib.rs"
required-features = ["derive"]
