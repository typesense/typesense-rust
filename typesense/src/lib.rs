#![deny(missing_docs)]
#![cfg_attr(docsrs, feature(doc_cfg))]

//! # Typesense
//!
//! Welcome to typesense, the rust library for the Typesense API.

#[cfg(feature = "openapi_client")]
pub use typesense_codegen as openapi_client;

mod client;
pub mod collection;
pub mod document;
mod error;
pub mod field;
pub mod transport;

pub use client::{keys, Client};
pub use error::{Result, TypesenseError};

#[cfg(feature = "typesense_derive")]
#[doc(hidden)]
pub use typesense_derive::*;
