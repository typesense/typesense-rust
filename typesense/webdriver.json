{
    "goog:chromeOptions": {
        "args": [
            "disable-web-security"
        ]
    }
}