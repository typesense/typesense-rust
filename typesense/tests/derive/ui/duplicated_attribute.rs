use typesense::Typesense;
use serde::{Serialize, Deserialize};
#[derive(Typesense, Serialize, Deserialize)]
struct Company {
    company_name: String,
    num_employees: i32,
    #[typesense(facet)]
    #[typesense(facet)]
    country_code: String,
}

fn main() {}
