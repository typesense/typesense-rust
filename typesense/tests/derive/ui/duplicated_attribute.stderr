error: #[typesense(facet)] repeated more than one time.
 --> $DIR/duplicated_attribute.rs:7:5
  |
7 | /     #[typesense(facet)]
8 | |     #[typesense(facet)]
9 | |     country_code: String,
  | |________________________^
